it/src/main/java/com/google/cloud/teleport/it/datastream/DatastreamResourceManager.java

```diff
@@ -0,0 +1,392 @@
+                                            DestinationConfig destinationConfig) {
+
+        LOG.info("Creating Stream {} in project {}.", streamId, projectId);
+
+        try {
+            CreateStreamRequest request =
+                    CreateStreamRequest.newBuilder()
+                            .setParent(LocationName.format(projectId, location))
+                            .setStreamId(streamId)
+                            .setStream(Stream
+                                    .newBuilder()
+                                    .setSourceConfig(sourceConfig)
+                                    .setDisplayName(streamId)
+                                    .setDestinationConfig(destinationConfig)
+                                    .setBackfillAll(Stream.BackfillAllStrategy.getDefaultInstance())
+                                    .build())
+                            .build();
+
+            Stream reference = datastreamClient.createStreamAsync(request).get();
+            createdStreamIds.add(streamId);
+
+            LOG.info("Successfully created Stream {} in project {}.", streamId, projectId);
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create stream. ", e);
+        }
+    }
+
+    public synchronized Stream updateStreamState(String streamId, Stream.State state) {
+
+        LOG.info("Updating {}'s state to {} in project {}.", streamId, state.name(), projectId);
+
+        try {
+            Stream.Builder streamBuilder = Stream.newBuilder()
+                    .setName(StreamName.format(projectId, location, streamId))
+                    .setState(state);
+
+            FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+            UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                    .setStream(streamBuilder)
+                    .setUpdateMask(fieldMaskBuilder)
+                    .build();
+
+            Stream reference = datastreamClient.updateStreamAsync(request).get();
+
+            LOG.info("Successfully updated {}'s state to {} in project {}.", streamId, state.name(), projectId);
+            return reference;
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatastreamResourceManagerException("Failed to update stream. ", e);
+        }
+    }
+
+    public synchronized Stream startStream(String streamId) {
+        LOG.info("Starting Stream {} in project {}.", streamId, projectId);
+        return updateStreamState(streamId, Stream.State.RUNNING);
+    }
+
+    public synchronized Stream pauseStream(String streamId) {
+        LOG.info("Pausing Stream {} in project {}.", streamId, projectId);
+        return updateStreamState(streamId, Stream.State.PAUSED);
+    }
+
+
+    public synchronized void cleanupAll() {
+        LOG.info("Cleaning up Datastream resource manager.");
+
+        try {
```

Comment:
Add a booelan to keep track of errors
```suggestion
        LOG.info("Cleaning up Datastream resource manager.");

        boolean producedError = false;
        try {
```